pyrene/src/components/DateTimePicker/DateTimeInput/DateTimeInput.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { forwardRef } from 'react';
+import clsx from 'clsx';
+
+import Icon from '../../Icon/Icon';
+import styles from './dateTimeInput.css';
+
+const allowedValueCheck = (valueToCheck:string) : boolean => (/^[0-9.: APM]*$/.test(valueToCheck));
+
+export interface DateTimeInputProps {
+  /**
+   * Boolean to toggle time display
+   */
+  dateOnly?: boolean,
+  dateValue?: string,
+  errorValue?: string,
+  handleOn?: (dateString: string, timeString: string) => void
+  disabled?: boolean,
+  label?: string,
+  name?: string,
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void, // Handle change function passed from react-datepicker
+  onClick?: () => void,
+  onFocus?: () => void,
+  range?: boolean,
+  setDateValue?: (value: string) => void,
+  setTimeValue?: (value: string) => void,
+  tabNum?: number,
+  timeValue?: string,
+  value?: string
+}
+
+const DateTimeInput = forwardRef(({
+  dateOnly = false,
+  dateValue = '',
+  errorValue = '',
+  handleOn,
+  disabled,
+  label,
+  name = '',
+  onChange = () => {},
+  onClick = () => {},
+  onFocus,
+  setDateValue = () => {},
+  setTimeValue = () => {},
+  tabNum,
+  timeValue = '',
+}:DateTimeInputProps, ref:React.Ref<HTMLInputElement>) => {
+
+  const handleDateOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const node = event.target as HTMLInputElement;
+
+    if (allowedValueCheck(node.value)) {
+      if (!dateOnly && node.value.length > 10) {
```

Comment:
You might have a `constant` for storing the length, like this 

```
const enum DATE_LENGTH {
  DATE_ONLY= 5;
  DATE_WITH_TIME =10,
}
```

Reply:
✅